Addded a warning when creating a bad PV.

RichardTysonIsis opened the pull request:
### Description of work

I added a warning in the block editor when you try to create a block containing `<pv name>:SP`, `<pv name>:SP:RBV`, `CS:SB:*` . This warning will not be called when an error message is to be displayed and does not stop the user from setting the block. I decided to set this warning in the block editing dialog as that is when the decision to create a block pointing to a certain PV is taken. See ticket for warning messages.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/2620

### Acceptance criteria

- [x] When you try pointing a block to a bad PV, a warning is issued.
- [x] A warning does not stop an error from being displayed (try for example `:SP::RBV` ) as the two colons trigger a error but `:SP:RBV` triggers a warning.
- [x] A warning does not stop the user from setting a block pointing to a bad PV.

### Unit tests

I added unit tests to check that the warning is triggered and that it contains the correct message.

### System tests

NA

### Documentation
None found.

---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Do the changes function as described and is it robust?
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

Tom-Willemsen commented on base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/editing/blocks/EditBlockDialog.java:
```diff
@@ -61,6 +62,20 @@ public void propertyChange(PropertyChangeEvent evt) {
+            for (ErrorMessageProvider model : viewModels) {
+                WarningMessage warning = model.getWarning();
+                if (warning.isWarning()) {
+                    setMessage(warning.getMessage(), 2);
```
I think it would be clearer to write this as `setMessage(warning.getMessage(), IMessageProvider.WARNING);`.

Tom-Willemsen commented on base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/editing/blocks/EditBlockDialog.java:
```diff
@@ -85,6 +100,10 @@ public EditBlockDialog(Shell parentShell, EditableBlock block, EditableConfigura
+
+		for (ErrorMessageProvider provider : viewModels) {
+            provider.addPropertyChangeListener("warning", warningListener);
+        }
```
This loop is exactly the same as the one above it, it would be neater to add both listeners in the same loop.

RichardTysonIsis commented:
The latest commit solved the bug Tom pointed, added unit tests to check for the related case and modified the code to take Tom's suggestions into account.
